src/libxrpl/basics/Number.cpp

```diff
@@ -250,57 +283,168 @@ Number::normalize()
         m /= 10;
         ++exponent_;
     }
-    mantissa_ = m;
-    if ((exponent_ < minExponent) || (mantissa_ < minMantissa))
+    if ((exponent_ < minExponent) || (m < minMantissa))
     {
-        *this = Number{};
+        mantissa_ = zero.mantissa_;
+        exponent_ = zero.exponent_;
+        negative = zero.negative_;
         return;
     }
+    // When using the largeRange, "m" needs fit within an int64, even if
+    // the final mantissa_ is going to end up larger to fit within the range.
+    // Cut it down here so that the rounding will be done while it's smaller.
+    //
+    // Example: 9,900,000,000,000,123,456 > 9,223,372,036,854,775,808,
```

Comment:
```suggestion
    // Example: 9,900,000,000,000,123,456 > 9,223,372,036,854,775,807,
```

the largest int64 is 9,223,372,036,854,775,80**7**